File: src/main/java/org/stormgears/utils/dsio/DummySwitch.java
package org.stormgears.utils.dsio;

import edu.wpi.first.wpilibj.command.Command;
import org.jetbrains.annotations.NotNull;

public class DummySwitch implements ISwitch {
	@Override
	public void whenFlipped(@NotNull SwitchControl.FlipListener listener) {

	}

	@Override
	public void whenPressed(Command command) {

	}

	@Override
	public void whileHeld(Command command) {

	}

	@Override
	public void whenReleased(Command command) {

	}

	@Override
	public void toggleWhenPressed(Command command) {

	}

	@Override
	public void cancelWhenPressed(Command command) {

	}

	@Override
	public void whenActive(Command command) {

	}

	@Override
	public void whileActive(Command command) {

	}

	@Override
	public void whenInactive(Command command) {

	}

	@Override
	public void toggleWhenActive(Command command) {

	}

	@Override
	public void cancelWhenActive(Command command) {

	}

	@Override
	public boolean get() {
		return false;
	}
}

File: src/main/java/org/stormgears/utils/dsio/DummySwitch.kt
package org.stormgears.utils.dsio

import edu.wpi.first.wpilibj.command.Command

open class DummySwitch : ISwitch {
	override fun whenFlipped(listener: (Boolean) -> Unit) {

	}

	override fun whenFlipped(listener: ISwitch.FlipListener) {

	}

	override fun whenPressed(command: Command) {

	}

	override fun whileHeld(command: Command) {

	}

	override fun whenReleased(command: Command) {

	}

	override fun toggleWhenPressed(command: Command) {

	}

	override fun cancelWhenPressed(command: Command) {

	}

	override fun whenActive(command: Command) {

	}

	override fun whileActive(command: Command) {

	}

	override fun whenInactive(command: Command) {

	}

	override fun toggleWhenActive(command: Command) {

	}

	override fun cancelWhenActive(command: Command) {

	}

	override fun get(): Boolean {
		return false
	}
}
